devices/Scd30/Scd30Sensor.cs

```diff
@@ -0,0 +1,419 @@
+﻿// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers.Binary;
+using System.IO.Ports;
+using Iot.Device.Scd30.Entities;
+using UnitsNet;
+
+namespace Iot.Device.Scd30
+{
+    /// <summary>
+    /// Sensirion SCD30 Sensor Module (CO2, humidity, and temperature sensor).
+    /// </summary>
+    /// <remarks>
+    /// Most of the SCD30 documentation below is derived or copied from the original documentation
+    /// at https://sensirion.com/media/documents/D7CEEF4A/6165372F/Sensirion_CO2_Sensors_SCD30_Interface_Description.pdf .
+    /// </remarks>
+    public class Scd30Sensor
+    {
+        private const byte Scd30ModbusAddress = 0x61;
+
+        private readonly SerialPort _serial;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Scd30Sensor" /> class to control the SCD30 via <see cref="SerialPort"/>. The BaudRate, DataBits, Parity, StopBits and timeouts will be configured automatically to interact with the SCD30.
```

Comment:
Maybe, but it would take away the opportunity for the developer to use the underlying SerialPort and configure other things on it, such as `InvertSignalLevels` or other values that could be PCB-specific. I'd rather take the SerialPort instance, configure it as required (to the knowledge of the library) and leave the rest in the hands of the developer. Not dumbing it down too much.

Besides, this opens the opportunity to introduce a `I2cDevice` overload if anyone ever implements the I2C protocol this device also supports (and clock stretching would work properly).

Reply:
OK and fair enough. Please then add this in the README for InvertSignalLevels as an option in the setup.